src/client/entity-editor/common/sort-name-field.js

```diff
@@ -0,0 +1,145 @@
+/*
+ * Copyright (C) 2016  Ben Ockmore
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, write to the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+
+import {Button, Input} from 'react-bootstrap';
+
+import React from 'react';
+import ValidationLabel from '../common/validation-label';
+
+/**
+ * Removes all period characters (dots) from the input string, returning a new
+ * string.
+ *
+ * @param {String} name the input string to strip
+ * @returns {String} the string with dots removed
+ */
+function stripDot(name) {
+	return name.replace(/\./g, '');
+}
+
+function makeSortName(name) {
```

Comment:
We'll want to give this code a more thorough review at some point. It may make the most sense to make it a specifiable parameter, since different entity types (and even different form field values) will require different behaviors.